### PR TITLE
update to DW 1.0

### PR DESCRIPTION
**Pull request by qinfchen**
- use DW 1.0
- java 8
- obtain context path from application context

### CODE REVIEW

**Comment by qinfchen** [comment]
@jmcampanini

File: build.gradle [hunk 1]
```diff
@@ -32,7 +32,5 @@ subprojects {
     apply plugin: 'com.palantir.baseline-eclipse'
     apply plugin: 'com.palantir.baseline-findbugs'
     apply plugin: 'com.palantir.baseline-idea'
-
-    sourceCompatibility = 1.7
```
Comment (jmcampanini) [inline build.gradle:36]: why not set this to 1.8? or is that the default for gradle?

File: dropwizard-index-page/src/main/java/com/palantir/indexpage/IndexPageBundle.java [hunk 2]
```diff
@@ -80,27 +76,9 @@ public void run(IndexPageConfigurable configuration, Environment environment) th
 
     private static void addIndexPageServlet(
             IndexPageConfigurable configuration, Environment environment, String indexPagePath, Set<String> mappings) {
-        String contextPath = getContextPath(configuration);
+        String contextPath = environment.getApplicationContext().getContextPath();
```
Comment (jmcampanini) [inline dropwizard-index-page/src/main/java/com/palantir/indexpage/IndexPageBundle.java:79]: SOOO MUCH NICER

File: dropwizard-index-page/src/test/java/com/palantir/indexpage/IndexPageServletTests.java [hunk 3]
```diff
@@ -38,7 +39,7 @@
     private final RequestDispatcher mockRequestDispatcher = mock(RequestDispatcher.class);
 
     @Before
-    @SuppressWarnings("unchecked")
```
Comment (jmcampanini) [inline dropwizard-index-page/src/test/java/com/palantir/indexpage/IndexPageServletTests.java:41]: what is no longer unchecked?

File: build.gradle (hunk 1 above)
Comment (qinfchen) [reply build.gradle:36]: yep default is 1.8

File: dropwizard-index-page/src/test/java/com/palantir/indexpage/IndexPageServletTests.java (hunk 3 above)
Comment (qinfchen) [reply dropwizard-index-page/src/test/java/com/palantir/indexpage/IndexPageServletTests.java:41]: i moved it to the class level to suppress the same warning from another test

File: build.gradle (hunk 1 above)
Comment (jmcampanini) [reply build.gradle:36]: cool. dunno if we want to be explicit and set 1.8. i'm indifferent. removing may be the best.

File: dropwizard-index-page/src/test/java/com/palantir/indexpage/IndexPageServletTests.java (hunk 3 above)
Comment (jmcampanini) [reply dropwizard-index-page/src/test/java/com/palantir/indexpage/IndexPageServletTests.java:41]: coo

**Comment by jmcampanini** [comment]
lgtm 👍

**Pull request merged**